src/mono/wasm/runtime/http.ts

```diff
@@ -41,14 +57,61 @@ export function http_wasm_abort_response(res: ResponseExtension): void {
     }
 }
 
+export function http_wasm_readable_stream_controller_enqueue(controller: ReadableByteStreamControllerExtension, bufferPtr: VoidPtr, bufferLength: number): void {
+    if (bufferLength === 0) {
+        controller.close();
+    } else {
+        // the bufferPtr is pinned by the caller
+        const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
+        const copy = view.slice() as Uint8Array;
+        if (controller.byobRequest) {
+            const v = controller.byobRequest.view as Uint8Array;
+            v.set(copy);
+            controller.byobRequest.respond(bufferLength);
+        } else {
+            controller.enqueue(copy);
+        }
+    }
+    controller.__resolve_controller();
+}
+
+export function http_wasm_readable_stream_controller_error(controller: ReadableByteStreamControllerExtension, error: Error): void {
+    controller.__reject_fetch(error);
+    controller.error(error);
+    controller.__resolve_controller();
+}
+
+export function http_wasm_fetch_stream(url: string, header_names: string[], header_values: string[], option_names: string[], option_values: any[], abort_controller: AbortController, pull: (controller: ReadableByteStreamControllerExtension, desired_size: number) => void): Promise<ResponseExtension> {
+    return new Promise((resolve_fetch, reject_fetch) => {
+        const body = new ReadableStream({
+            type: "bytes",
+            autoAllocateChunkSize: 65536,
+            pull(controller: ReadableByteStreamController) {
```

Comment:
> Maybe you need to change signature to Func<JSObject,int,Task> and await it here.

I would love to but the js import generator can't do that.
```
error SYSLIB1072: The type 'System.Func<System.Runtime.InteropServices.JavaScript.JSObject, int, System.Threading.Tasks.Task>' is not supported by source-generated JavaScript interop. The generated source will not handle marshalling of parameter 'pull'.
```
https://github.com/dotnet/runtime/blob/08b66559ed06d4d97fb09dfa0d20163f0865deef/src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSGeneratorFactory.cs#L270-L277

Reply:
> > Maybe you need to change signature to Func<JSObject,int,Task> and await it here.
> I would love to but the js import generator can't do that.

That's a shame, sorry about that. I spent few hours today trying quick hack to support it, because I thought Task of void is not really generic type and it would be easy hack.

But it needs to be done properly. That means support for generic types with generic type arguments. In generator as well as in the runtime. There are quite few useful other types, like `Task<byte[]>` which would be good to have.